javaslang/src/main/java/javaslang/collection/Array.java

```diff
@@ -567,6 +569,26 @@ private Array(Object[] delegate) {
     }
 
     @Override
+    public java.util.List<T> asJava() {
+        return JavaConverters.asJava(this, IMMUTABLE);
+    }
+
+    @Override
+    public Array<T> asJava(Consumer<? super java.util.List<T>> action) {
```

Comment:
this notion has several advantages:

* j.u.List is scoped / does not escape
* fluent API
* we benefit from inference, i.e. not full qualified names required on name clashes between standard Java and Javaslang:

```Java
List.of(1, 2, 3)
    .asJavaMutable(list -> {
        // hah!
        list.add(4);
    })
    .filter(i -> i % 2 == 0); // = List(2, 4)
```

More specifically:

```Java
javaslang.collection.List<Integer> numbers = List.of(1, 2, 3);

// = List(2, 4)
javaslang.collection.List<Integer> filtered = numbers.asJavaMutable(list -> {
    // hah!
    list.add(4);
}).filter(i -> i % 2 == 0);

// numbers is still List(1, 2, 3)!
```